src/apps/competitions/models.py

```diff
@@ -23,6 +28,107 @@ def __str__(self):
     def bundle_dataset(self):
         return CompetitionCreationTaskStatus.objects.get(resulting_competition=self).dataset
 
+    def check_future_phase_submissions(self):
+        """
+        Checks for if we need to migrate current phase submissions to next phase.
+        """
+        phases = self.phases.all()
+
+        if len(phases) == 0:
+            return
+
+        try:
+            current_phase = phases.get(status='Current')
+            next_phase = phases.get(status='Next')
+            final_phase = phases.get(status='Final')
+
+        except Phase.DoesNotExist:
+            return
+
+        # Making sure current_phase or next_phase is not None
+        if current_phase is None or next_phase is None or current_phase is final_phase:
+            return
+
+        logger.info("Checking for needed migrations on competition pk=%s, current phase: %s, next phase: %s" %
```

Comment:
One thing that is annoying on v1.5 is I see these messages all the time for competitions already migrated. Would be nice if these competitions _never_ ran the check again... some kind of `has_been_migrated` flag?

Reply:
Resolved in latest push